htdocs/product/class/product.class.php

```diff
@@ -4681,7 +4681,7 @@ public function getChildsArbo($id, $firstlevelonly = 0, $level = 1)
 			while ($rec = $this->db->fetch_array($res)) {
 				if (!empty($alreadyfound[$rec['rowid']])) {
 					dol_syslog(get_class($this).'::getChildsArbo the product id='.$rec['rowid'].' was already found at a higher level in tree. We discard to avoid infinite loop', LOG_WARNING);
-					continue;
+					//continue;
```

Comment:
If we remove this, you remove an important protection.
If A is linked to B and B to C and C to A, this will generates problem. In such a case, it means data are corrupted in database, because we dont want to have a relation parent-child with data with a child that has child that is already a parent of him.
So if we have in database  A -> B -> C -> A, we cant to return only 
A -> B -> C   (supposing that link C -> A is an error in database, it should not happen but if this happen, the function must deal it)